test/matrix_multiply.jl

```diff
@@ -44,6 +44,13 @@
         m = @SMatrix [1 2 3 4]
         v = @SVector [1, 2]
         @test @inferred(v*m) === @SMatrix [1 2 3 4; 2 4 6 8]
+
+        # Outer product
+        v2 = zeros(SVector{3, Int})
+        @test v2 * v2' === zeros(SMatrix{3, 3, Int})
+
+        v3 = zeros(SVector{3, Complex{Int}})
+        @test v3 * v3' === zeros(SMatrix{3, 3, Complex{Int}})
```

Comment:
In this case, I'd test the end behaviour - the tests will be less brittle that way, and the call chain from public interface to private implementation is not complicated to follow.

In general it's a tradeoff:
* Testing the implementation details can let you write more specific tests for smaller blocks of code.  When things break, isolating the problem is easier.
* Testing the interface lets you refactor the implementation without breaking the tests.  Refactoring very detailed implementation tests can be a lot of work, and isn't much fun.